### DISCUSSION TITLE
Does BirdNET perform detection before classification?

### DISCUSSION THREAD

**Discussion by shnchh** [birdnet-team/BirdNET-Analyzer#490] [Q&A]
Hi, just some quick clarifications:
1. During inference, does BirdNET perform detection to segment the audio clips into potential detected calls? Or does it split the whole audio into 3 sec segments and perform detection on all these segments.
2. When training a custom classifier, does BirdNET perform data augmentation on the new training set? If not, would it be recommended to do so?
3. I understand that BirdNET can append the new classifier's labels onto BirdNET's classifier. I am wondering how it works if there is a label/bird that already exists in BirdNET's classification layer labels, will the labels be deconflicted and merged before training? 
4. What is the point of having a negative samples class in training since it won't be used for binary classification?

Thank you!

**Comment by Josef-Haupt** [comment]
1. We split the signal into 3s segments and perform classification on each one.
2. If you check the options in the gui or CLI, we will perform mixup and/or upsamling of minority classes. But no further augmentations. Dataaugmentation can be helpful in cases of low samples per class, but it is not always necessary, depends on the performance of your custom classifier and your requirements.
3. If you are using "append" for your custom classifier, the classes of your training data will be appended to the existing BirdNET output. It is a stupid matrix operation with no further logic behind it. So if you have a class that is already present in BirdNET in your training data, you can check both values, the one from the BirdNET and the appended value, which might be different, because the last layer will have different weights after your training.
4. Negative samples can for example be useful, to train the classifier to seperate similar sounds. There might be a machine that sounds like a bird.

Was that helpful to you?

**Reply by shnchh** [reply]
That was very helpful, thank you!
